src/transaction/TransactionController.ts

```diff
@@ -952,6 +990,21 @@ export class TransactionController extends BaseController<
     }
     return latestIncomingTxBlockNumber;
   }
+
+  /**
+   * Resolves the locally stored transactions with the blockchain or etherscan. Then updated TransactionController State
+   * @param address - public address to resolve transactions against
+   * @param method - use StateReconsileMethod enum to specifiy method of resolution. Default will be the blockchain
+   */
+  async transactionStateReconciler (address: string, method?: StateReconsileMethod) {
+    
+    // Leverage fetchAll() or queryTransactionStatuses() to resolve locally stored txs
+
+    // If the transaction reported on the blockchain/etherscan has reach an end state 
+    // (cancelled = 'cancelled', confirmed = 'confirmed', failed = 'failed', rejected = 'rejected') 
```

Comment:
Remove the polling logic and leverage the existing methods (etherscan/blockchain)

Resolve chain and etherscan logic

- cancelled ("dropped" reported by)
- failed - but never reached the chain
- add status for failed off chain

Reply:
Added into comments and new status field